pint/registry.py

```diff
@@ -230,10 +262,30 @@ def __init__(
         fmt_locale: Optional[str] = None,
         non_int_type: NON_INT_TYPE = float,
         case_sensitive: bool = True,
+        cache_folder: Union[str, pathlib.Path, None] = None,
```

Comment:
This would be `Optional[Union[str, pathlib.Path]]`. BTW, I was trying a tool to convert all annotations into modern ones (i.e. List -> list, etc). I will do it after the release.

Reply:
Union will be fine, it's Tue with three options, it works well